### ISSUE TITLE
Objects365_v1 category mismatch

### ISSUE THREAD

**Issue by huangqiu15444** [ultralytics/ultralytics#22099]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Why do I encounter category mismatch issues when parsing the `objects365_train_segm.json` file? For example, I found that the category of car is 9 instead of 6.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @huangqiu15444, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you unblocked quickly—an Ultralytics engineer will also follow up here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For your Objects365_v1 category mismatch question, please also include:
-Exact file name/version (e.g., objects365_train_segm.json) and a small JSON snippet showing the categories and an affected annotation
-The code you used to parse categories/annotations (CLI command or Python snippet)
-The exact Ultralytics version and environment details
-Attached minimal sample (few lines) that reproduces the mismatch

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by huangqiu15444** [comment]
#load json data 
json_path = 'objects365_train_segm.json'
json_data = json.load(open(json_path)) # dict_keys(['annotations', 'images', 'licenses', 'type', 'info', 'categories'])

label_dict = defaultdict(list)
save_dict = defaultdict(list)
cnt = 0
cls_set = set()
print("process annotations, ano to imageId list ...")
for ano_info in tqdm(json_data['annotations']):
    cls_idx = ano_info['category_id'] - 1 # 
    seg_info = ano_info['segmentation']
    img_id = ano_info['image_id']
    seg_info.append(cls_idx)
    label_dict[img_id].append(seg_info) # 
    #print(len(seg_info))
    cls_set.add(cls_idx)
  
for img_info in tqdm(json_data['images']):
    file_name = img_info['file_name']
    img_id = img_info['id']
    height = img_info['height']
    width = img_info['width']

    #img_dict[img_id] = img_info
    if img_id not in label_dict:
        continue
    
    image = cv2.imread(os.path.join('./images_src/train', file_name))
    height,width,ch = image.shape 

    #print(len(ano_info['segmentation']), "---", ano_info['segmentation'])
    save_flag = False
    for ano_info in label_dict[img_id]:
        cls_idx = int(ano_info.pop())
        seg_obj = np.array(ano_info[0]).reshape((-1,2)) 

        # show labels
        pts = seg_obj.reshape((-1,1,2)).astype(np.int32)
        color = (random.randint(0,255), random.randint(0,255), random.randint(0,255))
        cv2.polylines(image, [pts], isClosed=True, color=color, thickness=2)
        text = str(cls_idx)
        org = (pts[0,0,0], pts[0,0,1])
        fontFace = cv2.FONT_HERSHEY_SIMPLEX
        fontScale = 1
        thickness = 1
        cv2.putText(image, text, org, fontFace, fontScale, color, thickness, cv2.LINE_AA)

    cv2.imwrite('visul.jpg', image)

**Comment by huangqiu15444** [comment]
#load json data 
json_path = 'objects365_train_segm.json'
json_data = json.load(open(json_path)) # dict_keys(['annotations', 'images', 'licenses', 'type', 'info', 'categories'])

label_dict = defaultdict(list)
save_dict = defaultdict(list)
cnt = 0
cls_set = set()
print("process annotations, ano to imageId list ...")
for ano_info in tqdm(json_data['annotations']):
    cls_idx = ano_info['category_id']# 
    seg_info = ano_info['segmentation']
    img_id = ano_info['image_id']
    seg_info.append(cls_idx)
    label_dict[img_id].append(seg_info) # 
    #print(len(seg_info))
    cls_set.add(cls_idx)
  
for img_info in tqdm(json_data['images']):
    file_name = img_info['file_name']
    img_id = img_info['id']
    height = img_info['height']
    width = img_info['width']

    #img_dict[img_id] = img_info
    if img_id not in label_dict:
        continue
    
    image = cv2.imread(os.path.join('./images_src/train', file_name))
    height,width,ch = image.shape 

    #print(len(ano_info['segmentation']), "---", ano_info['segmentation'])
    save_flag = False
    for ano_info in label_dict[img_id]:
        cls_idx = int(ano_info.pop())
        seg_obj = np.array(ano_info[0]).reshape((-1,2)) 

        # show labels
        pts = seg_obj.reshape((-1,1,2)).astype(np.int32)
        color = (random.randint(0,255), random.randint(0,255), random.randint(0,255))
        cv2.polylines(image, [pts], isClosed=True, color=color, thickness=2)
        text = str(cls_idx)
        org = (pts[0,0,0], pts[0,0,1])
        fontFace = cv2.FONT_HERSHEY_SIMPLEX
        fontScale = 1
        thickness = 1
        cv2.putText(image, text, org, fontFace, fontScale, color, thickness, cv2.LINE_AA)

    cv2.imwrite('visul.jpg', image)
    import pdb;pdb.set_trace()

**Comment by Y-T-G** [comment]
I guess you need to ask this in YOLOE repo because the author of YOLOE created it

**Comment by huangqiu15444** [comment]
> I guess you need to ask this in YOLOE repo because the author of YOLOE created it

Thank you for your reply. I've found the corresponding `Objects365v1.yaml` file.

**Comment by glenn-jocher** [comment]
Great—using the Objects365 YAML is the right move; note that COCO-style `category_id` in `objects365_train_segm.json` is dataset-defined and doesn’t equal the 0‑based YOLO index, so map IDs to the YAML `names` (or let the YAML’s converter handle it). For a manual remap:

```python
import json, yaml

y = yaml.safe_load(open("Objects365v1.yaml"))  # use the YAML you found
name2idx = {v.lower(): k for k, v in y["names"].items()}
j = json.load(open("objects365_train_segm.json"))
id2name = {c["id"]: c["name"].lower() for c in j["categories"]}
id2yolo = {cid: name2idx[id2name[cid]] for cid in id2name}
# then: cls_idx = id2yolo[ann["category_id"]]
```

For reference, see the Objects365 guide and its YAML with the full class order in the repository: [Objects365 dataset guide](https://docs.ultralytics.com/datasets/detect/objects365/) and [Objects365.yaml (repo)](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/Objects365.yaml).

**Comment by liuyuchen796** [comment]
Have you used the weights trained by objects365

**Comment by glenn-jocher** [comment]
We don’t publish Objects365‑pretrained YOLO11 checkpoints; we typically fine‑tune from COCO `yolo11*.pt`. If you have your own O365‑trained weights, use them with the same O365 YAML used for training to avoid class‑order issues, e.g.:

```bash
yolo detect val data=path/to/Objects365*.yaml model=path/to/your_objects365_weights.pt
```

For available base checkpoints to fine‑tune from, see the [YOLO11 models](https://docs.ultralytics.com/models/yolo11/).
